programs/bpf_loader/src/lib.rs

```diff
@@ -1927,7 +1981,50 @@ mod tests {
         assert_eq!(
             TransactionError::InstructionError(1, InstructionError::InvalidAccountData),
             bank_client
-                .send_and_confirm_message(&[&mint_keypair, &program_keypair], message)
+                .send_and_confirm_message(
+                    &[&mint_keypair, &program_keypair, &upgrade_authority_keypair],
+                    message
+                )
+                .unwrap_err()
+                .unwrap()
+        );
+
+        // Mismatched buffer and program authority
```

Comment:
Could you also add a test that upgrades/deploys fail when buffer authority is `None`?